ui/src/components/admin/faqs/FaqGrid.js

```diff
@@ -34,24 +67,201 @@ export default function FaqGrid(props) {
+        // Reset FAQ editing panel when multiple FAQs are selected, else show top FAQ
+        selectedNodes.length === 1
+          ? setSelected(selectedNodes[0].data)
+          : resetHandler();
+
+        // Handle publish/unpublish buttons
+        const selectedFaqsIsPublished = selectedNodes.map(
+          (faq) => faq.data.isPublished
+        );
+        setToPublish(
+          selectedFaqsIsPublished.every((isPublished) => isPublished === false)
+        );
+        setToUnpublish(
+          selectedFaqsIsPublished.every((isPublished) => isPublished === true)
+        );
+      } else {
+        // Reset FAQ editing panel when no FAQs are selected
+        resetHandler();
+        // Hide publish/unpublish buttons when no FAQs are selected
+        setToPublish(false);
+        setToUnpublish(false);
+      }
+    }
+  };
+
+  // Functions to handle FAQ reordering
+  useEffect(() => {
+    if (api) {
+      api.setSuppressRowDrag(!isReorderingFaqs);
+      if (isReorderingFaqs) {
+        setReorderFaqColumns();
+        setReorderFaqFilters();
+      } else {
+        setReorderFaqColumns();
+        setReorderFaqFilters();
+      }
+    }
+  }, [isReorderingFaqs, api]);
+
+  // Set the column visibility when reordering FAQs
+  const setReorderFaqColumns = () => {
+    if (colApi) {
+      colApi.setColumnVisible('pageTopic', !isReorderingFaqs);
+      colApi.setColumnVisible('createdBy', !isReorderingFaqs);
+      colApi.setColumnVisible('lastUpdatedBy', !isReorderingFaqs);
+      colApi.setColumnVisible('isPublished', !isReorderingFaqs);
+      colApi.setColumnVisible('isDeleted', !isReorderingFaqs);
+    }
+  };
+
+  // Set the filters when reordering FAQs
+  const setReorderFaqFilters = () => {
+    setShowPublishedOnly(isReorderingFaqs);
+    setPageTopicFilter(isReorderingFaqs ? pageTopicList[0] : '');
   };
 
+  // FAQ Publishing & Reordering Functions
+  // Handles FAQ publishing/unpublishing
+  const publishHandler = async () => {
+    if (api) {
+      const selectedFaqs = api.getSelectedNodes();
+
+      if (selectedFaqs && selectedFaqs.length) {
+        if (toPublish) {
+          setToPublishList(selectedFaqs);
+          selectedFaqs.forEach((faq) => {
+            faq.setDataValue('isPublished', true);
+          });
+          resetHandler();
+          setIsReorderingFaqs(true);
+        } else if (toUnpublish) {
+          setToUnpublishList(selectedFaqs);
+          selectedFaqs.forEach((faq) => {
+            faq.setDataValue('isPublished', false);
+          });
+          resetHandler();
+          setIsReorderingFaqs(true);
+        }
+      } else {
+        toast({
+          description: 'No FAQs selected',
+          status: 'error',
+          duration: 3000,
+        });
+      }
+    }
+  };
+  // Enables FAQ reordering
+  const enableFaqReordering = () => {
+    setShowDeleted(false);
+    setIsReorderingFaqs(true);
+  };
+  // Cancels FAQ publishing/reordering
+  const cancelHandler = () => {
```

Comment:
oh i get it now why.
can you just do this? right now you're forcing the behaviour and it kinda looks like hard coding to me. same applies for line 152
```
  const cancelHandler = () => {
    toPublishList.forEach((faq) => {
      faq.setDataValue('isPublished', !toPublish);
    });
    setToPublish(!toPublish);
    setIsReorderingFaqs(false);
  };
```